### PR TITLE
Devel

### PR DESCRIPTION
**Pull request by Danfoa**
## Summary

This PR merges the recent `devel` work into `main` and advances `symm-learning` from `0.4.1` to `0.7.2`.

The branch adds new equivariant linear algebra and statistics functionality, fixes cache and backpropagation issues in core equivariant modules, and expands the transformer stack with positional-attention variants and a new conditional transformer model. It also updates the public package surface, documentation, and test coverage to match the new APIs.

## Main Changes

- Add new linear algebra utilities for equivariant projections and maps, including `equiv_orthogonal_projection`, `equiv_orthogonal_projection_coefficients`, `equiv_linear_map`, and structured isotypic projection metadata.
- Improve numerical and runtime behavior in core math/stats code by fixing covariance computation, stabilizing `irrep_radii` gradients, and ensuring cached tensors stay on the correct dtype and device.
- Introduce `eModule` infrastructure and cache-handling fixes so equivariant modules preserve correct behavior across deepcopy, load/save, and training or evaluation transitions.
- Extend running statistics with `eEMAStats` covariance tracking in hom-space degrees of freedom and a fix for its backpropagation behavior.
- Expand the neural module stack with positional attention backends, including rotary and additive positional embeddings, and move transformer components into `symm_learning.nn`.
- Add the new `CondTransformer` model built on the positional-attention transformer layers.
- Update package initialization and version resolution, including source-version lookup from `pyproject.toml` and singleton-preserving deepcopy handling for `escnn` group and representation objects.
- Refresh documentation and examples for isotypic decomposition, equivariant linear maps, and representation theory, and add profiling/runtime comparison utilities.

## Testing

This PR includes test updates and additions across:
- `test/test_linalg.py`
- `test/test_stats.py`
- `test/test_modules.py`
- `test/test_models.py`
- `test/test_utils_rep_theory.py`

### CODE REVIEW

File: symm_learning/nn/activation.py [hunk 1]
```diff
@@ -89,7 +88,7 @@ def __init__(
             bias=bias,
             add_bias_kv=add_bias_kv,
             add_zero_attn=add_zero_attn,
-            batch_first=batch_first,
+            batch_first=True,
```
[the remark this answers is not in the document]
Comment (Danfoa) [reply symm_learning/nn/activation.py:91]: In this repository wu will always default to batch_first equals true, so we need to remove the presence of these parameter, and always call batch_first on torch modules if any

File: symm_learning/nn/transformer/transformer.py [hunk 2]
```diff
@@ -0,0 +1,614 @@
...
+            output = self.norm(output)
+        return output
+
+
+def _get_activation_module(activation: str | Callable) -> torch.nn.Module:
+    """Resolve a feed-forward activation to an :class:`torch.nn.Module`."""
+    if isinstance(activation, torch.nn.Module):
+        return activation
+    if callable(activation):
+        return torch.nn.GELU() if activation is F.gelu else torch.nn.ReLU()
```
[the remark this answers is not in the document]
Comment (Danfoa) [reply symm_learning/nn/transformer/transformer.py:584]: Yes respect the user passed activation module. Implement these changes
